infobind/cache.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infobind
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser"
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/parser/terror"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/sqlexec"
+	log "github.com/sirupsen/logrus"
+)
+
+const defaultBindCacheSize = 5
+
+// bindData store the basic bind info and bindSql astNode.
+type bindData struct {
+	bindMeta
+	ast ast.StmtNode
+}
+
+// bindCache holds a bindDataMap, key:origin sql hash value: bindData slice.
+type bindCache struct {
+	Cache map[string][]*bindData
+}
+
+// Handler hold a atomic bindCache.
+type Handler struct {
+	bind atomic.Value
+}
+
+// HandleUpdater use to update the bindCache.
+type HandleUpdater struct {
```

Comment:
1.
``` go
// BindCacheUpdater is used to update the global bindCache.
type BindCacheUpdater struct {
	ctx            sessionctx.Context

	parser         *parser.Parser
	lastUpdateTime types.Time
	globalHandler  *Handler
}
```
2. We need to add some comments to describe when and how we update the global bindCache.

Reply:
ok